File: src/app/core/models/booking.models.ts
/**
 * Booking-related models
 * Extracted from class-booking.service.ts and shared.models.ts
 */

export interface ClassBooking {
    id: string;
    studentId: string;
    teacherId: string;
    subject: string;
    date: Date;
    startTime: string;
    endTime: string;
    status: 'Pending' | 'Confirmed' | 'Cancelled' | 'Completed';
    paymentStatus: 'Pending' | 'Paid' | 'Failed' | 'Refunded';
    classType: 'Personal_1_1' | 'Group' | 'OneTime' | 'Recurring'; // Extending existing types
    mode: 'ONLINE' | 'IN_PERSON';
    locationOrLink: string;
    price: number;
    durationMinutes: number;
    bookingGradeLevel: string;
    recurringDays?: string[];
    notes?: string;
    meetingLink?: string;
    createdAt: Date;
    updatedAt: Date;
}

export interface BookingRequest {
    teacherId: string;
    subject: string;
    date: Date;
    startTime: string;
    endTime: string;
    classType: 'Personal_1_1' | 'Group' | 'OneTime' | 'Recurring';
    mode: 'ONLINE' | 'IN_PERSON';
    bookingGradeLevel: string;
    notes?: string;
}

export interface BookingResponse {
    success: boolean;
    message: string;
    booking?: ClassBooking;
}

File: Backend/ClassBooking.API/Models/BookingModels.cs
namespace ClassBooking.API.Models
{
    public class ClassBooking
    {
        public string Id { get; set; } = string.Empty;
        public string StudentId { get; set; } = string.Empty;
        public string TeacherId { get; set; } = string.Empty;
        public string Subject { get; set; } = string.Empty;
        public DateTime Date { get; set; }
        public string StartTime { get; set; } = string.Empty;
        public string EndTime { get; set; } = string.Empty;
        public string Status { get; set; } = "Pending"; // Pending, Confirmed, Cancelled, Completed
        public string ClassType { get; set; } = "OneTime"; // OneTime, Recurring
        public List<string>? RecurringDays { get; set; }
        public string? Notes { get; set; }
        public string? MeetingLink { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }

    public class BookingRequest
    {
        public string TeacherId { get; set; } = string.Empty;
        public string Subject { get; set; } = string.Empty;
        public DateTime Date { get; set; }
        public string StartTime { get; set; } = string.Empty;
        public string EndTime { get; set; } = string.Empty;
        public string ClassType { get; set; } = "OneTime";
        public string? Notes { get; set; }
    }

    public class BookingResponse
    {
        public string Id { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public string Message { get; set; } = string.Empty;
    }
}
